src/main/java/teammates/ui/pagedata/InstructorFeedbackResultsPageData.java

```diff
@@ -698,7 +698,10 @@ private void buildSectionPanelsForForAjaxLoading(List<String> sections) {
 
         InstructorFeedbackResultsSectionPanel sectionPanel = new InstructorFeedbackResultsSectionPanel(
                 Const.DEFAULT_SECTION, Const.NO_SPECIFIC_SECTION, true);
-        sectionPanels.put(Const.DEFAULT_SECTION, sectionPanel);
+
+        if (bundle.getTeamsInSectionFromRoster(Const.DEFAULT_SECTION).size() > 0) {
```

Comment:
In addition, this won't work for some cases. If you take a careful look at how `getTeamsInSectionFromRoster` is implemented. You will notice that only students are used to generate to map. That is, if we have instructors/general feedback as recipients, you won't display the default section while there is content in the default section.

Reply:
Hmm, thanks a lot for pointing that out. My initial line of thought is that all sections have teams because teams are compulsory during course enrollment.

What other scenarios will there be no teams but there are content? Is there a good way to determine that using data from `bundle`?

Reply:
What @xpdavid is saying is correct. If I add a question whose recipient is `Nobody specefic`, then in case of results by `RGQ` and `RQG`, the `No Section panel` is not empty.

But this arises another question, should the panel which shows the `responses to Nobody specefic` be called `No specefic Section` or `No specefic recipient`.
This is why the naming of panel has been changed before 
https://github.com/TEAMMATES/teammates/issues/7633#issuecomment-369217282

I think we need to figure out what was the original intention of that panel and it was meant for showing what exactly?
In the current release 6.3.0, it shows 
- all responses of students not belonging to any section
- responses given to recepient as `general`

I think its current behaviour is inconsistent with its naming, which is causing all the confusion

Reply:
Agree.
The description is correctly described the things that is display inside `No specific Section`:
> - all responses of students not belonging to any section
> - responses given to recepient as general

I would say it is just `responses given to recipient as general` doesn't fit the `No specific Section`. `No specific recipient` should be more suitable.

Reply:
@tshradheya thanks for giving your input as well. I think it is important to clarify what responses `No Specific Section` will have and when it will be non-empty. If not there might be cases where we remove the section but it should have responses.

Reply:
Can I clarify what type of `responses` these are `responses given to recipient as general`?

Reply:
@Crphang You can choose the feedback path as `xxxx -> Nobody specific (For general class feedback)`

Reply:
'No specific recipient' panels are at a inner level than 'No specific section' panels. That's because grouping based on section happen first before grouping by receiver.
It seems 'No specific recipient' panels are currently shown as a dash. I'm fine with that.
![image](https://user-images.githubusercontent.com/1673303/36936092-6cda8300-1f3b-11e8-9b73-7573456c2983.png)

On an unrelated note, @Crphang the PR title should follow the issue title exactly, unless the PR is only a partial fix or has no matching issue.

Reply:
@Crphang When there is no actual recipient of the response. If this is what you were asking? :P 
![image](https://user-images.githubusercontent.com/12574756/36936094-7ad5c924-1f3b-11e8-96f3-a99f057e725a.png)

@xpdavid 
With the current logic of the panel, renaming it to any of the two `No specific section` or `No specific recepient` will be wrong.

- If renamed to `No specific recipient` it will still show responses which have an actual recipient [ this is when no sections are there, so **all** responses are shown in the panel]
- If kept as it is, it is wrong because it shows responses which have recepient as `general feedback` which is not what the name says

Reply:
Prof @damithc Thanks for the clarification.

Got one doubt though. 
Case: All students in some section. The feedback session has 1 question with Feedback path `students-> nobody specefic`

Now if Instructor views result in `RGQ` or `RQG` mode the results of this 1 question appear inside the panel `No specific section` when the students actually are in some section. 
This is inconsistent with what the panel says, right?

Reply:
Yes, that should not be the case. Probably a bug.

Reply:
@xpdavid After discussing with @tshradheya, there are a couple of solution to this, but each with it's own downside.

1. Include a new `No Specific Recipient` panel that is specifically for `RGQ` and `RQG`. And remove the `general` feedback for instructors from `No Specific Recipient`.
    - We do a frontend check to remove `No Specific Recipient` section `onDocumentLoad` instead of `onClick`, if the view type is *not* `RGQ` or `RQG`. This removes the possibility of the similar bug to occur.
    - This is clean because it creates a very clear separation of `No Specific Recipient` and `No Specific Section` but is a involved solution. Requires changing what goes into `No Specific Sections`, and adding a respective section and section handler.

1. We add another check to check whether there are any responses that is meant as `General Feedback`, and we remove the `No Specific Section` section if there is no such response. However, the bug still persists for view type that is `GQR` or `GRQ` if there are such responses. Because `No Specific Section` is empty for `GQR` and `GRQ` but is non-empty for `RQG` and `RGQ`.
    - Pros: This fix is faster. But doesn't solve the problem completely, but can mitigate it for most cases.

Please advice!

Reply:
I think there are some misunderstanding here.

For @tshradheya question, the behaviour is correct to me. When you group the responses by `RGQ` or `RQG` mode. You would put recipient first, then question/givers blablabla. Therefore, the sections you are talking about are where the recipients should be put to. If the recipient is `No body specific` (general feedback), then it should belong to `No Specific Section` and then belong to `No Specific Recipient`.

@Crphang To me, the first solution is a "hacking" solution. The hidden section`<div>` will "kill" other contributors when they try to fix something >_<. We can do explicitly check (second method) when we add default panel. You can introduce viewType when you `buildSectionPanelsForForAjaxLoading` to see whether we want to add default panel or not.

Reply:
@xpdavid, will attempt that